lib/src/element_type.dart

```diff
@@ -23,7 +23,11 @@ class ElementType {
 
   bool get isParameterizedType {
     if (_type is FunctionType) {
-      return (_type as FunctionType).typeFormals.isNotEmpty;
+      final FunctionType functionType = _type;
+      final Iterable<DartType> typeArguments = []
+          ..addAll(functionType.typeArguments)
+          ..addAll(functionType.typeFormals.map((f) => f.type));
+      return typeArguments.isNotEmpty;
     } else if (_type is ParameterizedType) {
```

Comment:
More constructively, I would try to fix it like this:

```dart
    bool get isParameterizedType {
      var type = _type; // NOTE: type promotion now works
      if (type is FunctionType) {
        // NOTE: you could factor out a common function if the performance cost
        // of building the list is a concern. It would be my code below but skipping
        // the .map(...).toList() part. 
        return typeArguments.isNotEmpty;
      } // ...
    }
    // ...
    List<ElementType> get typeArguments {
      var type = _type;
      if (type is FunctionType) { 
        List<DartType> typeArguments;
        if (type.element is FunctionTypeAliasElement && type.typeFormals.isEmpty) {
          // TODO(jmesserly): simplify check above; we should have a way
          // to find instantiated typedefs without consulting the element.
          // Also, it will not work if we support typedefs declared inside classes.
          typeArguments = type.typeArguments;
        } else {
          typeArguments = type.typeFormals;
        }
        return typeArguments.map(_getElementTypeFrom).toList();
      } // ...
    }
```